Assets/Scripts/Resources/ResourceSpawner.cs

```diff
@@ -64,16 +64,19 @@ void Start()
 
 	void OnDrawGizmos()
 	{
-		Transform trans = GetComponent<Transform>();
-		Gizmos.DrawIcon( trans.position + Vector3.up * 0.5f, "ResourceSpawner.png", true );
+		Transform transform = GetComponent<Transform>();
+
+		Gizmos.matrix = transform.localToWorldMatrix;
+
+		Gizmos.DrawIcon( transform.position + Vector3.up * 0.5f, "ResourceSpawner.png", true );
```

Comment:
So the weird thing I was noticing was that `Gizmos.matrix` wasn't affecting `DrawIcon()`, so the cube was appearing relative to the object, but all of the text icons were showing up at the origin (or `Vector3.up * 0.5f). Adding the transform's position back in fixed it, but I found it weird.